### PR TITLE
Acceptance Test for LoadBalancer and Monitor

### PR DESCRIPTION
**Pull request by Pushpalatha555**
**Validated the Acceptance Test for LB and Monitor**

LB
NO-LB

LB-DATA
LB-M

### CODE REVIEW

File: acc-testcases/data-sources/load_balancer.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+acc:
+- config: |
+    name = "tf_loadbalancer"
```
Comment (manjunath-batakurki) [inline acc-testcases/data-sources/load_balancer.yaml:3]: I hope you have cleaned up everything except this resource. Can you add a validation here to check the ID and any other fields if any.

File: acc-testcases/data-sources/load_balancer_monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+acc:
+- config: |
+    name = "tf_lb_monitor"
```
Comment (manjunath-batakurki) [inline acc-testcases/data-sources/load_balancer_monitor.yaml:3]: same here

File: acc-testcases/data-sources/nsx-tier1.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+acc:
+- config: |
+    name = "nsx_tier1_gateway"
```
Comment (manjunath-batakurki) [inline acc-testcases/data-sources/nsx-tier1.yaml:3]: same here

File: acc-testcases/resources/load_balancer.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    group_access {
+      all = true
+    }
+    config {
+      admin_state = true
+      size = "SMALL"
+      log_level = "INFO"
+      tier1_gateways  = "/infra/tier-1s/26cdb82e-0057-4461-ad4d-cddd61d77b1f"
+    }
+
```
Comment (manjunath-batakurki) [inline acc-testcases/resources/load_balancer.yaml:17]: Can we add some validations here that is unique ?

File: acc-testcases/resources/lb_monitor.yaml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    name  =  "$(icmp_name)"        
+    description  = "ICMP_MONITOR update using tf"
+    type = "LBIcmpMonitorProfile"
+    icmp_monitor {
+      fall_count = 30
+      interval =  50 
+      monitor_port = 8
+      rise_count = 3
+      timeout = 15
+      data_length = 32
```
Comment (manjunath-batakurki) [inline acc-testcases/resources/lb_monitor.yaml:60]: Can you add at least one negative testcase ? Please refer instance testcase(template Error validation)

File: acc-testcases/data-sources/load_balancer.yaml (hunk 1 above)
Comment (Pushpalatha555) [reply acc-testcases/data-sources/load_balancer.yaml:3]: yes

File: acc-testcases/data-sources/nsx-tier1.yaml (hunk 3 above)
Comment (manjunath-batakurki) [reply acc-testcases/data-sources/nsx-tier1.yaml:3]: add a validation to see if the provider_id is returned here.

**Review by manjunath-batakurki — COMMENTED** [review]
Please attach the acceptance test results here. Validate them offline individually by running one after the other.

File: acc-testcases/data-sources/load_balancer_monitor.yaml (hunk 2 above)
Comment (Pushpalatha555) [reply acc-testcases/data-sources/load_balancer_monitor.yaml]: done

File: acc-testcases/data-sources/nsx-tier1.yaml (hunk 3 above)
Comment (Pushpalatha555) [reply acc-testcases/data-sources/nsx-tier1.yaml]: done

File: acc-testcases/resources/lb_monitor.yaml (hunk 5 above)
Comment (Pushpalatha555) [reply acc-testcases/resources/lb_monitor.yaml:60]: done

**Comment by Pushpalatha555** [comment]
> Please attach the acceptance test results here. Validate them offline individually by running one after the other.

updated the screenshots above for the acceptance test results

File: acc-testcases/resources/load_balancer_monitor-monitor_err.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  rand_name_1: tf_HTTP_MONITOR_%rand_int
+acc:
+- config: |
+    lb_id = 262 
+    name  =  "$(http_name)"      
+    description  = "HTTP_MONITOR creating using tf"
+    type = "LBHttpMonitorProfile"
+    http_monitor {
+      fall_count = 8
+      interval =  10
```
Comment (manjunath-batakurki) [inline acc-testcases/resources/load_balancer_monitor-monitor_err.yaml:11]: how is this a negative scenario ? i see the type and the body both are same. what are u trying to check here ?

Comment (Pushpalatha555) [reply acc-testcases/resources/load_balancer_monitor-monitor_err.yaml:11]: here I am trying to validate for the name value line 2 having variable rand_name_1 for the name value but inside config line 6 variable value is http_name but its not valid one.
now added one sceanrios like the type missmatch for the body.

**Review by manjunath-batakurki — APPROVED** [review]

**Review by ravikumar29524 — APPROVED** [review]

**Pull request merged**